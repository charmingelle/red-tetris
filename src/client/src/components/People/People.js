import React from 'react';
import { connect } from 'react-redux';
import './People.css';

const getColor = (personId, myId) => (personId === myId ? '#c00' : '#f9f9f9');

const PeopleInner = ({ people, myId }) => (
  <div className="lobby-people">
    <div className="lobby-people-label">Players</div>
    <div className="lobby-people-list">
      {Object.keys(people)
        .sort((id1, id2) => people[id2].score - people[id1].score)
        .map(id => (
          <div
            key={id}
            className="lobby-person-name"
            style={{ color: getColor(id, myId) }}
          >{`${people[id].name} - ${people[id].score}`}</div>
        ))}
    </div>
  </div>
);

const mapStateToProps = ({ people, myId }) => ({ people, myId });

export const People = connect(mapStateToProps)(PeopleInner);
