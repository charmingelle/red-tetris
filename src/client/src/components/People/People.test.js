import React from 'react';
import { People } from './People';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

test(`Two people, Anna's screen`, () => {
  const store = mockStore({
    socket: null,
    myId: 'id-0',
    myName: null,
    myRoom: null,
    people: {
      'id-0': { name: 'Anna', score: 0 },
      'id-1': { name: 'Olya', score: 3 },
    },
    rooms: {},
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <People />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

test(`Two people, Olya's screen`, () => {
  const store = mockStore({
    socket: null,
    myId: 'id-1',
    myName: null,
    myRoom: null,
    people: {
      'id-0': { name: 'Anna', score: 0 },
      'id-1': { name: 'Olya', score: 3 },
    },
    rooms: {},
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <People />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
