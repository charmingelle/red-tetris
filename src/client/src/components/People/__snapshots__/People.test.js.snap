// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Two people, Anna's screen 1`] = `
<div
  className="lobby-people"
>
  <div
    className="lobby-people-label"
  >
    Players
  </div>
  <div
    className="lobby-people-list"
  >
    <div
      className="lobby-person-name"
      style={
        Object {
          "color": "#f9f9f9",
        }
      }
    >
      Olya - 3
    </div>
    <div
      className="lobby-person-name"
      style={
        Object {
          "color": "#c00",
        }
      }
    >
      Anna - 0
    </div>
  </div>
</div>
`;

exports[`Two people, Olya's screen 1`] = `
<div
  className="lobby-people"
>
  <div
    className="lobby-people-label"
  >
    Players
  </div>
  <div
    className="lobby-people-list"
  >
    <div
      className="lobby-person-name"
      style={
        Object {
          "color": "#c00",
        }
      }
    >
      Olya - 3
    </div>
    <div
      className="lobby-person-name"
      style={
        Object {
          "color": "#f9f9f9",
        }
      }
    >
      Anna - 0
    </div>
  </div>
</div>
`;
